### PR TITLE
Adding Sorting Map Objects

### PR DESCRIPTION
**Pull request by Cyperghost**
Adding the function to sort map objects by distance

### CODE REVIEW

**Comment by Grover-c13** [comment]
Thanks! could you please add the license header to the new files please

**Comment by Cyperghost** [comment]
Yes of course

**Comment by Grover-c13** [comment]
It looks like some of the files tabs have changed, can you check its passing the checkstyle when using gradle build?

**Comment by FabianTerhorst** [comment]
You have to add the Style xml to your ide

**Comment by Cyperghost** [comment]
Yes, i have

File: src/main/java/com/pokegoapi/api/gym/Gym.java [hunk 1]
```diff
@@ -30,11 +30,12 @@
 import com.pokegoapi.exceptions.LoginFailedException;
 import com.pokegoapi.exceptions.RemoteServerException;
 import com.pokegoapi.main.ServerRequest;
+import com.pokegoapi.util.MapPoint;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class Gym {
+public class Gym implements MapPoint{
```
Comment (FabianTerhorst) [inline src/main/java/com/pokegoapi/api/gym/Gym.java:38]: add space between Point and {

File: src/main/java/com/pokegoapi/api/map/Point.java [hunk 2]
```diff
@@ -16,10 +16,11 @@
 package com.pokegoapi.api.map;
 
 import POGOProtos.Map.SpawnPointOuterClass;
+import com.pokegoapi.util.MapPoint;
 import lombok.Getter;
 import lombok.Setter;
 
-public class Point {
+public class Point implements MapPoint{
```
Comment (FabianTerhorst) [inline src/main/java/com/pokegoapi/api/map/Point.java:23]: look above

File: src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java [hunk 3]
```diff
@@ -50,7 +52,7 @@
  * The type Catchable pokemon.
  */
 @ToString
-public class CatchablePokemon {
+public class CatchablePokemon implements MapPoint{
```
Comment (FabianTerhorst) [inline src/main/java/com/pokegoapi/api/map/pokemon/CatchablePokemon.java:55]: look above

**Pull request merged**